snapshots/storage/metastore.go

```diff
@@ -117,34 +117,29 @@ func (ms *MetaStore) WithTransaction(ctx context.Context, writable bool, fn Tran
 		return err
 	}
 
-	var result *multierror.Error
+	var result []error
 	err = fn(ctx)
 	if err != nil {
-		result = multierror.Append(result, err)
+		result = append(result, err)
 	}
 
 	// Always rollback if transaction is not writable
 	if err != nil || !writable {
 		if terr := trans.Rollback(); terr != nil {
 			log.G(ctx).WithError(terr).Error("failed to rollback transaction")
 
-			result = multierror.Append(result, fmt.Errorf("rollback failed: %w", terr))
+			result = append(result, fmt.Errorf("rollback failed: %w", terr))
 		}
 	} else {
 		if terr := trans.Commit(); terr != nil {
 			log.G(ctx).WithError(terr).Error("failed to commit transaction")
 
-			result = multierror.Append(result, fmt.Errorf("commit failed: %w", terr))
+			result = append(result, fmt.Errorf("commit failed: %w", terr))
 		}
 	}
 
-	if err := result.ErrorOrNil(); err != nil {
+	if err := errors.Join(result...); err != nil {
 		log.G(ctx).WithError(err).Debug("snapshotter error")
-
-		// Unwrap if just one error
-		if len(result.Errors) == 1 {
-			return result.Errors[0]
-		}
```

Comment:
I also removed this `Unwrap` since most cases don't have such `Unwrap` logic unless specifically needed.

Only such case I found is the code below (to handle `cleanupErr` separately).

https://github.com/containerd/containerd/blob/fb786b2f17b2c7d7ea5d9c9009d980bc03379952/pkg/cri/sbserver/sandbox_run.go#L256-L266